core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java

```diff
@@ -402,11 +404,10 @@ public Map<String,Pair<UUID,String>> getScanServers() {
       try {
         final var zLockPath = ServiceLock.path(root + "/" + addr);
         ZcStat stat = new ZcStat();
-        byte[] lockData = ServiceLock.getLockData(getZooCache(), zLockPath, stat);
-        if (lockData != null) {
-          String[] fields = new String(lockData, UTF_8).split(",", 2);
-          UUID uuid = UUID.fromString(fields[0]);
-          String group = fields[1];
+        ServiceLockData sld = ServiceLock.getLockData(getZooCache(), zLockPath, stat);
```

Comment:
I like the idea of creating the ServiceLockData to abstract away some of this implementation with more expressive APIs. I think that could be done independently of adding the group concept. Doing that first may make it easier to see just the changes required to add a tserver group feature, separately from the code organization work.